app/src/main/java/com/github/se/eduverse/repository/TodoRepository.kt

```diff
@@ -0,0 +1,143 @@
+package com.github.se.eduverse.repository
+
+import android.util.Log
+import com.github.se.eduverse.model.Todo
+import com.github.se.eduverse.model.TodoStatus
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+open class TodoRepository(private val db: FirebaseFirestore) {
+  private val collectionPath = "todos"
+
+  fun init(onAuthStateChanged: (userId: String?) -> Unit) {
+    Firebase.auth.addAuthStateListener {
+      if (it.currentUser != null) {
+        onAuthStateChanged(it.currentUser!!.uid)
```

Comment:
Instead of forcing !! on it.currentUser.uid, consider using ?.let to handle nullability safely. This will help avoid any potential NullPointerException if the user data unexpectedly becomes null